src/User/MysqlToggleStatusModifierService.php

```diff
@@ -80,174 +95,103 @@ private function toggleStatusUnset( $toggleStatus )
-     */
-    private function generateQueryBuilderForGroupToggleDelete()
-    {
-        $queryBuilder = new QueryBuilder( $this->connection );
-        $queryBuilder
-            ->delete( 'group_policy' )
-            ->where( 'toggle_id = ?' )
-            ->andWhere( 'group_id = ?' );
-        return $queryBuilder;
+        return $identityColumn;
     }
 
     /**
-     * @param QueryBuilder $queryBuilder
-     * @param string $toggleIdentifier
-     * @param string $groupIdentifier
-     * @param bool $isActive
-     */
-    private function updateToggle( QueryBuilder $queryBuilder, $toggleIdentifier, $groupIdentifier, $isActive )
-    {
-        $queryBuilder
-            ->setParameter( 0, $isActive )
-            ->setParameter( 1, $toggleIdentifier )
-            ->setParameter( 2, $groupIdentifier );
-        $queryBuilder->execute();
-    }
-
-    /**
-     * @param string $toggleIdentifier
-     * @param string $userIdentifier
+     * @param StringifyableTable $policyTable
+     * @param int $toggleIdentifier
+     * @param string $identity
+     * @return bool
      */
-    private function tryInsertElseUpdateUserToggleToActiveState( $toggleIdentifier, $userIdentifier )
+    private function policyRecordExists( StringifyableTable $policyTable, $toggleIdentifier, $identity )
     {
-        try {
-            $this->insertActiveUserActivatedToggle( $toggleIdentifier, $userIdentifier );
-        } catch ( \Exception $e ) {
-            $queryBuilder = $this->generateQueryBuilderForToggleUpdate( '`user_policy`', 'user_id' );
-            $this->updateToggle( $queryBuilder, $toggleIdentifier, $userIdentifier, true );
-        }
+        $identityColumn = $this->getIdentityColumnByPolicyTable( $policyTable );
+        $numberOfPolicyRecords = $this->connection->createQueryBuilder()
+                                                  ->select( "COUNT(toggle_id)" )
+                                                  ->from( (string)$policyTable )
+                                                  ->where( "toggle_id = ?" )
+                                                  ->andWhere( $identityColumn . " = ?" )
+                                                  ->setParameter( 0, $toggleIdentifier )
+                                                  ->setParameter( 1, $identity )
+                                                  ->execute()->fetchColumn();
+        return $numberOfPolicyRecords > 0;
     }
 
     /**
-     * @param string $toggleIdentifier
-     * @param string $groupIdentifier
+     * @param StringifyableTable $policyTable
+     * @param int $toggleIdentifier
+     * @param string $identity
      * @param bool $isActive
+     * @return int
+     * @throws \Doctrine\DBAL\DBALException
      */
-    private function tryInsertElseUpdateGroupToggleToAGivenState( $toggleIdentifier, $groupIdentifier, $isActive )
+    private function replacePolicyRecord( StringifyableTable $policyTable, $toggleIdentifier, $identity, $isActive )
```

Comment:
Haha @luos so ok, you're right about the base problem. I was confused for a little bit if we really want to break the DB independency of the repo, but then I realised this is literally called labs-mysql, so... :D

Since this is a MySQL-only operation, we don't have support for it neither in the connection object, nor with the query builder. And yeah, we don't use ORM, that was a decision made during earlier phases of labs development. What do you mean by why I don't use the table object?